packages/core/src/components/Typography/FontSize/FontSizeData.tsx

```diff
@@ -0,0 +1,61 @@
+import { Box, Text } from '@chakra-ui/react'
+import { TFontSizeVariant } from '@core/types'
+
+// max font size in px or em/rem
+const maxFontSizePx = 48
+const maxFontSizeEmRem = maxFontSizePx / 16
+
+// return boolean font size is too large for preview
+function isFontSizeTooLarge(fontSizeData: TFontSizeVariant) {
+  if (fontSizeData.unit === 'px') {
+    return fontSizeData.value > maxFontSizePx
+  }
+
+  if (fontSizeData.unit === 'rem' || fontSizeData.unit === 'em') {
+    return fontSizeData.value > maxFontSizeEmRem
+  }
+
+  return false
+}
+
+// return font size for preview default to 1rem if too large
+function normalizeFontSize(fontSizeData: TFontSizeVariant) {
+  if (fontSizeData.unit === 'px') {
+    return isFontSizeTooLarge(fontSizeData) ? '1rem' : `${fontSizeData.value}px`
+  }
+
+  if (fontSizeData.unit === 'rem' || fontSizeData.unit === 'em') {
+    return isFontSizeTooLarge(fontSizeData)
+      ? '1rem'
+      : `${fontSizeData.value}${fontSizeData.unit}`
+  }
+
+  return '1rem'
+}
+
+export function FontSizeData({
```

Comment:
nit: Would prefer to leave original naming of `FontSizeRow` for both the component + file name. I'm worried `FontSizeData` will be a confusing search result.